# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/IPCIdentityCredential.ipdlh
# Commit: ccd96b99aea2
# Full Hash: ccd96b99aea28e8214e585b864d3fb2245bac142
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-06-07 08:31:23
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing failures complaining about fedcm-no-registered-idps.https.html.
#   
#   This reverts commit ba33457e0cbac24a8b7d963c45138c62e92a74ae.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 4bfadfc33207 -r ccd96b99aea2 dom/credentialmanagement/identity/IPCIdentityCredential.ipdlh
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/credentialmanagement/identity/IPCIdentityCredential.ipdlh	Fri Jun 06 21:35:18 2025 +0300
@@ -0,0 +1,31 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include "mozilla/dom/PermissionMessageUtils.h";
+[RefCounted] using class nsIPrincipal from "nsIPrincipal.h";
+
+namespace mozilla {
+namespace dom {
+
+// Most of these fields are taken directly from mozilla::dom::IdentityCredentialInit
+// Others are:
+//    identityProvider: the principal that created the IdentityCredential this represents.
+//    infoExpiresAt: PR_Now() + the mozilla::dom::IdentityCredentialInit's infoExpiresAfter, in ms
+struct IPCIdentityCredential
+{
+  nsString id;
+  nsString? token;
+  nsCString? name;
+  nsCString? iconURL;
+  nsCString[] effectiveOrigins;
+  nsCString? effectiveQueryURL;
+  nsCString? effectiveType;
+  uint64_t? infoExpiresAt;
+  nullable nsIPrincipal identityProvider;
+};
+
+}
+}